### PR TITLE
New graph model (composable, extension based), plus early alpha of Inspector

### PR DESCRIPTION
**Pull request by kolektiv**
Tentative pull request to test the waters/build...

### CODE REVIEW

**Comment by kolektiv** [comment]
Right, this is quite broken, I'll take a look at it shortly!

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "ui/inspector"]
+	path = ui/inspector
+	url = https://github.com/freya-fs/freya.ui.inspector.git
```
Comment (panesofglass) [inline .gitmodules]: I find it strange to have a submodule dependency on a project that depends on other projects in this repo. What uses this dependency? Also, this is very likely the reason the build is broken.

File: build.fsx [hunk 2]
```diff
@@ -97,12 +97,33 @@ let freya =
...
+                  { Name = "Freya.Machine.Extensions.Http"
+                    Dependencies =
+                        [ Package "FSharp.Core"
+                          Package "Aether"
+                          Local "Freya.Core"
+                          Local "Freya.Machine"
+                          Local "Freya.Types.Http"
+                          Local "Freya.Types.Language"
+                          Local "Freya.Types.Uri" ] }
+                  { Name = "Freya.Machine.Extensions.Http"
```
Comment (panesofglass) [inline build.fsx]: This is the same name as the extensions above, isn't it? Is this meant to be `Freya.Machine.Extensions.Cors`?

File: src/Freya.Inspector/Content.fs [hunk 3]
```diff
@@ -44,23 +49,38 @@ let private getCss =
...
         including defaults
         mediaTypesSupported html
-        handleOk getHtml } |> compileFreyaMachine
+        handleOk getHtml } |> Machine.toPipeline
+
+let private js =
+    freyaMachine {
+        including defaults
+        mediaTypesSupported js
+        handleOk getJs } |> Machine.toPipeline
```
Comment (panesofglass) [inline src/Freya.Inspector/Content.fs]: I like this name much, much better! Thanks!

File: .gitmodules (hunk 1 above)
Comment (kolektiv) [inline .gitmodules]: I'm not totally sure what you mean here - the submodule dependency is on a separate repo, which has no dependency back to this - at least, only an implicit one! This submodule provides the content which is embedded in Freya.Inspector to provide a UI. I don't think any kind of cycle is present here, if that's what you mean, but I may just be missing your meaning :)

File: build.fsx (hunk 2 above)
Comment (kolektiv) [inline build.fsx]: Yes, I should think so! I'll fix that shortly!

File: src/Freya.Inspector/Content.fs (hunk 3 above)
Comment (kolektiv) [inline src/Freya.Inspector/Content.fs]: It's definitely better I agree!

**Comment by kolektiv** [comment]
Hi @panesofglass the build is getting better, but there seems to be some issue with the PDB stuff. Possibly one we've seen before? I know you got that running last time, can you remember if it was this issue?

**Comment by panesofglass** [comment]
I believe it's because the Raw GitHub url used in the `SourceLink` commands points to the `freya-fs` org and does not account for the fact that the source is actually in your `kolektiv` account.

**Comment by kolektiv** [comment]
Ah, yes that would make sense. Not too sure what to do about that! Could we disable that for PR builds?

**Pull request merged**